ui/pages/add-token/add-token.component.js

```diff
@@ -211,16 +220,20 @@ class AddToken extends Component {
   }
 
   handleCustomDecimalsChange(value) {
-    const customDecimals = value.trim();
-    const validDecimals =
-      customDecimals !== null &&
-      customDecimals !== '' &&
-      customDecimals >= MIN_DECIMAL_VALUE &&
-      customDecimals <= MAX_DECIMAL_VALUE;
-    let customDecimalsError = null;
+    const inValidDecimals = !(
+      value >= MIN_DECIMAL_VALUE && value <= MAX_DECIMAL_VALUE
+    );
 
-    if (!validDecimals) {
-      customDecimalsError = this.context.t('decimalsMustZerotoTen');
+    let customDecimals;
+    let customDecimalsError = null;
+    if (value) {
+      customDecimals = Number(value.trim());
```

Comment:
nit: could we maybe define this with numerical coercion at the top of the function? Then we can use strict comparison operators in the decimal validation check and fail early if needed if `value` is falsey/not a number